connectors/sources/salesforce.py

```diff
@@ -0,0 +1,1443 @@
+        )
+
+        # ensure string, remove Nones, and remove whitespace
+        bodies = [str(x[-1]).strip() for x in time_body_pairs if x[-1] is not None]
+
+        # finally filter out any empty strings, join and return
+        return "\n\n".join(filter(None, bodies))
+
+    def _collect_case_participant_ids_emails_and_names(self, case):
+        ids = []
+        emails = []
+        names = []
+
+        participants = [
+            case.get("Owner", {}) or {},
+            case.get("CreatedBy", {}) or {},
+        ]
+
+        case_comments = case.get("CaseComments", {}) or {}
+        participants.extend(self._collect_created_by(case_comments))
+
+        email_messages = case.get("EmailMessages", {}) or {}
+        participants.extend(self._collect_created_by(email_messages))
+
+        for email in email_messages.get("records", []):
+            names.append(email.get("FromName"))
+            emails.append(email.get("FromAddress"))
+            emails.extend(str(email.get("ToAddress", "")).split(";"))
+            emails.extend(str(email.get("CcAddress", "")).split(";"))
+            emails.extend(str(email.get("BccAddress", "")).split(";"))
+
+        feeds = case.get("Feeds", []) or []
+        for feed in feeds:
+            participants.append(feed.get("CreatedBy", {}))
+
+            feed_comments = feed.get("FeedComments", {}) or {}
+            participants.extend(self._collect_created_by(feed_comments))
+
+        for participant in participants:
+            ids.append(participant.get("Id"))
+            emails.append(participant.get("Email"))
+            names.append(participant.get("Name"))
+
+        ids = self._format_list(ids)
+        emails = self._format_list(emails)
+        names = self._format_list(names)
+
+        return ids, emails, names
+
+    def _collect_created_by(self, sobject):
+        created_by_list = []
+
+        records = sobject.get("records", [])
+        for record in records:
+            created_by_list.append(record.get("CreatedBy"))
+
+        return created_by_list
+
+    def _format_list(self, unformatted):
+        return sorted(
+            set(filter(None, [str(x).strip() for x in unformatted if x is not None]))
+        )
+
+
+class SalesforceDataSource(BaseDataSource):
+    """Salesforce"""
+
+    name = "Salesforce"
+    service_type = "salesforce"
+
+    def __init__(self, configuration):
+        super().__init__(configuration=configuration)
+        self.base_url = BASE_URL.replace("<domain>", configuration["domain"])
+        self.salesforce_client = SalesforceClient(
+            configuration=configuration, base_url=self.base_url
+        )
+        self.doc_mapper = SalesforceDocMapper(self.base_url)
+
+    def _set_internal_logger(self):
+        self.salesforce_client.set_logger(self._logger)
+
+    @classmethod
+    def get_default_configuration(cls):
+        return {
+            "client_id": {
+                "label": "Client ID",
```

Comment:
Suggestion: add RCF `validations` to configuration fields, [RCF docs](https://github.com/elastic/connectors-python/blob/main/docs/DEVELOPING.md#rich-configurable-fields)

Reply:
What unique validations do you think these require?

Reply:
Of the possible custom validation types (`'less_than' | 'greater_than' | 'list_type' | 'included_in' | 'regex'`) I don't see anything that we could validate on here.

Reply:
Ahh you are right, I somehow linked `required` property to live in `validations`. 👍